lib/grpc/GrpcService.ts

```diff
@@ -17,15 +19,40 @@ class GrpcService {
     try {
       const getInfoResponse = await this.service.getInfo();
       const response = new wallirpc.GetInfoResponse();
-      response.setVersion(getInfoResponse['version']);
-      response.setProtocolversion(getInfoResponse['protocolversion']);
-      response.setBlocks(getInfoResponse['blocks']);
-      response.setTimeoffset(getInfoResponse['timeoffset']);
-      response.setConnections(getInfoResponse['connections']);
-      response.setProxy(getInfoResponse['proxy']);
-      response.setDifficulty(getInfoResponse['difficulty']);
-      response.setTestnet(getInfoResponse['testnet']);
-      response.setRelayfee(getInfoResponse['relayfee']);
+      response.setVersion(getInfoResponse.version);
+
+      const getLndInfo = ((lndInfo: LndInfo): wallirpc.LndInfo => {
+        const lnd = new wallirpc.LndInfo();
+        const { version, chainsList, blockheight, uris, error } = lndInfo;
+
+        if (lndInfo.channels) {
+          const channels = new wallirpc.LndChannels();
+          channels.setActive(lndInfo.channels.active);
+          channels.setPending(lndInfo.channels.pending);
+          lndInfo.channels.inactive ? channels.setInactive(lndInfo.channels.inactive) : undefined;
+          lnd.setLndchannels(channels);
+        }
+
+        version ? lnd.setVersion(version) : undefined;
```

Comment:
Why not just `lnd.setVersion(version)`? Is there an error when setting the version to `undefined`?

Reply:
yes, there is `lnd.setVersion(version)` cant accept `undefined` .